process/block/sovereignChainBlock.go

```diff
@@ -916,17 +921,31 @@ func (scbp *sovereignChainBlockProcessor) createOutGoingMiniBlockData(outGoingOp
 		OutGoingOperations: outGoingOperationsData,
 	})
 
-	// TODO: We need to have a mocked transaction with this hash to be saved in storage and get rid of following warnings:
-	// 1. basePreProcess.createMarshalledData: tx not found hash = bf7e...
-	// 2. basePreProcess.saveTransactionToStorage  txHash = bf7e... dataUnit = TransactionUnit error = missing transaction
-	// Task for this: MX-14716
 	return &block.MiniBlock{
 		TxHashes:        outGoingOpHashes,
 		ReceiverShardID: core.MainChainShardId,
 		SenderShardID:   scbp.shardCoordinator.SelfId(),
 	}, outGoingOperationsHash
 }
 
+func (scbp *sovereignChainBlockProcessor) addOutGoingTxToPool(outGoingOp *sovCore.OutGoingOperation) {
+	tx := &transaction.Transaction{
+		GasLimit: scbp.economicsData.ComputeGasLimit(
+			&transaction.Transaction{
+				Data: outGoingOp.Data,
+			}),
+		GasPrice: scbp.economicsData.MinGasPrice(),
+		Data:     outGoingOp.Data,
+	}
+
+	scbp.dataPool.Transactions().AddData(
+		outGoingOp.Hash,
+		tx,
+		tx.Size(),
+		fmt.Sprintf("%d_%d", core.SovereignChainShardId, core.MainChainShardId),
```

Comment:
You can use here `cacheId := process.ShardCacherIdentifier(core.SovereignChainShardId,  core.MainChainShardId)`

Reply:
+1

Reply:
I'm very much afraid this will cause a go linter issue which will say to put it directly without variable, but changed as request